pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -345,10 +407,32 @@ func jobNameHash(name string) string {
 	return hex.EncodeToString(hasher.Sum(nil))
 }
 
-func generateProwjob(rc injectingResolverClient, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, pr *v1.PullRequestUnderTest) (*prowv1.ProwJob, error) {
+func resolveCiopConfig(rc injectingResolverClient, baseCiop *api.Metadata, spec v1.ReleaseJobSpec) (*api.ReleaseBuildConfiguration, error) {
+	inject := &api.MetadataWithTest{
+		Metadata: api.Metadata{
+			Org:     spec.CIOperatorConfig.Org,
+			Repo:    spec.CIOperatorConfig.Repo,
+			Branch:  spec.CIOperatorConfig.Branch,
+			Variant: spec.CIOperatorConfig.Variant,
+		},
+		Test: spec.Test,
+	}
+
+	ciopConfig, err := rc.ConfigWithTest(baseCiop, inject)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get config from resolver: %w", err)
+	}
+
+	return ciopConfig, nil
+}
+
+type options struct {
+	labels map[string]string
+}
+
+func generateProwjob(ciopConfig *api.ReleaseBuildConfiguration, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, pr *v1.PullRequestUnderTest, mimickedJob string, extraOptions ...options) (*prowv1.ProwJob, error) {
```

Comment:
```suggestion
func generateProwjob(ciopConfig *api.ReleaseBuildConfiguration, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, pr *v1.PullRequestUnderTest, mimickedJob string, labels map[string]string) (*prowv1.ProwJob, error) {
```

Reply:
I added the `extraOptions` here in order to avoid passing nil maps for the case where there are no extra labels needed for the job, and also passing a struct here allows to be extended in every future need.